src/components/contract-details.tsx

```diff
@@ -25,16 +26,16 @@ const PluralizedItem: React.FC<BoxProps & { array: any[]; label: string }> = ({
   </Text>
 );
 
-export const ContractDetails = ({ contractId, contractInterface }) => {
+export const ContractDetails = ({ contractInterface, tx }) => {
   return (
     <Section flexShrink={0} mb="extra-loose" minWidth="200px" title="Contract details">
       <Box px="base">
         <Flex borderBottom={border()} py="base" justifyContent="space-between" alignItems="center">
           <Flex alignItems="center">
-            <ItemIcon size="48px" type="tx" txType="smart_contract" status="success" />
+            <ItemIcon size="48px" tx={tx} />
```

Comment:
@fbwoolf looking at this more -- this is not technically correct.

When on a contract-call page, we don't want to pass the contract-call transaction to the icon, we instead want to pass the parent contract information to this. maybe we can mock the tx object to be of the parent contract deploy, or fetch it (i think it's fetched already)

Reply:
Ok, I'll look at this. I thought we wanted the contract-call transaction here. Should be easy to change.